File: Codes/0557-reverse-words-in-a-string-iii.cpp
/*
 * @Author: LetMeFly
 * @Date: 2023-03-14 13:46:32
 * @LastEditors: LetMeFly
 * @LastEditTime: 2023-03-14 13:49:07
 */
#ifdef _WIN32
#include "_[1,2]toVector.h"
#endif

class Solution {
public:
    string reverseWords(string& s) {
        int begin = 0;
        for (int i = 0; i <= s.size(); i++) {
            if (i == s.size() || s[i] == ' ') {
                reverse(s.begin() + begin, s.begin() + i);
                begin = i + 1;
            }
        }
        return s;
    }
};

File: Codes/0557-reverse-words-in-a-string-iii.py
'''
Author: LetMeFly
Date: 2023-03-14 13:46:43
LastEditors: LetMeFly
LastEditTime: 2023-03-14 13:53:42
'''
class Solution:
    def reverseWords(self, s: str) -> str:
        return ' '.join(i[::-1] for i in s.split())